src/main/java/core/basesyntax/reportservices/ReportCreatorImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.reportservices;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ReportCreatorImpl implements ReportCreator {
+    @Override
+    public List<String> report(Map<String, Integer> transactions) {
+        List<String> report = new ArrayList<>();
+        report.add("fruit, quantity");
+        for (Map.Entry<String, Integer> entry : transactions.entrySet()) {
+            report.add(entry.getKey() + "," + entry.getValue());
```

Comment:
To improve the encapsulation, consider creating a separate method that formats the transaction entry into a report string. This way, if the format needs to change, you only need to update it in one place.